src/ngraph_builder.cc

```diff
@@ -374,6 +374,87 @@ static tf::Status TranslateAvgPoolOp(const tf::Node* op,
   return tf::Status::OK();
 }
 
+static tf::Status TranslateAvgPoolGradOp(const tf::Node* op, 
+                                     Builder::OpMap& ng_op_map) { 
+      
+  shared_ptr<ng::Node> ng_grad;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, nullptr, &ng_grad));
+
+  std::vector<tf::int32> tf_orig_input_shape_vec;
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(
+      op->attrs(), "_ngraph_avgpoolgrad_static_input_shape", &tf_orig_input_shape_vec));
+
+  std::vector<tf::int32> tf_strides;
+  std::vector<tf::int32> tf_ksize;
+  std::string tf_padding_type;
+  std::string tf_data_format;
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(), "strides", &tf_strides));
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(), "ksize", &tf_ksize));
+  TF_RETURN_IF_ERROR(
+      tf::GetNodeAttr(op->attrs(), "padding", &tf_padding_type));
+  TF_RETURN_IF_ERROR(
+      tf::GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
+
+  if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+    return tf::errors::InvalidArgument(
+        "AvgPoolGrad data format is neither NHWC nor NCHW");
+  }
+
+  bool is_nhwc = (tf_data_format == "NHWC");
+
+  NGRAPH_VLOG(3) << ng::join(tf_strides);
+  NGRAPH_VLOG(3) << ng::join(tf_ksize);
+  NGRAPH_VLOG(3) << tf_padding_type;
+  NGRAPH_VLOG(3) << tf_data_format;
+
+  ng::Shape ng_orig_input_shape;
+  for (int i = 0; i < tf_orig_input_shape_vec.size(); i++) {
+    ng_orig_input_shape.push_back(tf_orig_input_shape_vec[i]);
+  }
+
+  ng::Shape ng_forward_arg_shape(4);
+  ng::Strides ng_strides(2);
+  ng::Shape ng_image_shape(2);
+  ng::Shape ng_kernel_shape(2);
+
+  BatchedOpParamReshape(is_nhwc, ng_orig_input_shape, ng_forward_arg_shape);
+  BatchToNGraph(is_nhwc, ng_grad);
+  BatchedOpParamToNGraph(is_nhwc, tf_strides, ng_strides);
+  BatchedOpParamToNGraph(is_nhwc, ng_forward_arg_shape, ng_image_shape);
+  BatchedOpParamToNGraph(is_nhwc, tf_ksize, ng_kernel_shape);
```

Comment:
ng_forward_arg_shape is in NCHW format (line 420). if data format is NHWC, ng_image_shape will not get the right value (line 423). Similarly for line 424, Ksize is FH,FW,IC,OC, irrespective of data format, ng_kernel_shape is not a function of tf_ksize and is_nhwc.

Reply:
I am not sure I understand what you mean. But in `AvgPoolGrad` input `      ksize: A list of `ints` that has length `>= 4`.
        The size of the sliding window for each dimension of the input.
      strides: A list of `ints` that has length `>= 4`.
        The stride of the sliding window for each dimension of the input.

`